### PR TITLE
Add on uninstall link for clearing all extra data:

### PR DESCRIPTION
**Pull request by a-kom**
1) Fixed DI for ConfirmClearMenuForm.
2) Added uninstall validator and link in it for clearing extra data.
3) Added an action in MenuController for clearing menu extra data.

### CODE REVIEW

File: src/ProxyClass/MenuItemExtrasUninstallValidator.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
+<?php
+// @codingStandardsIgnoreFile
```
Comment (iVegas) [inline src/ProxyClass/MenuItemExtrasUninstallValidator.php]: Why should this file be ignored?

Comment (a-kom) [reply src/ProxyClass/MenuItemExtrasUninstallValidator.php]: It is autogenerated, you could check example in book core module.
And file isn't ignored - this is ingnorence of autogenerated file doc.

File: src/MenuItemExtrasUninstallValidator.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  public function __construct(Connection $connection, TranslationInterface $string_translation) {
+    $this->connection = $connection;
+    $this->stringTranslation = $string_translation;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($module) {
+    if ($module == 'menu_item_extras' && $this->hasExtraData()) {
```
Comment (iVegas) [inline src/MenuItemExtrasUninstallValidator.php]: `==` => `===`

Comment (a-kom) [reply src/MenuItemExtrasUninstallValidator.php]: Pls, recheck.

**Pull request merged**